src/nccl_ofi_net.c

```diff
@@ -966,26 +966,29 @@ static ncclResult_t ofi_process_cq(nccl_ofi_t *nccl_ofi_comp)
 			if (OFI_UNLIKELY(rc < 0)) {
 				NCCL_OFI_WARN("Unable to read from fi_cq_readerr. RC: %zd. Error: %s",
 					     rc,
-					     fi_cq_strerror(cq,
-						err_buffer.prov_errno,
-						err_buffer.err_data, NULL, 0));
+					     fi_strerror(-rc));
 				ret = ncclSystemError;
 				goto exit;
 			}
 
-			/* TODO: Add debug log to dump failed request details */
 			req = container_of(err_buffer.op_context,
 					   nccl_ofi_req_t, ctx);
 			req->state = NCCL_OFI_REQ_ERROR;
 			req->size = err_buffer.len;
+			NCCL_OFI_WARN("Request %p completed with error. RC: %d. Error: %s",
+					 req,
+					 err_buffer.err,
+				     fi_cq_strerror(cq,
+					err_buffer.prov_errno,
+					err_buffer.err_data, NULL, 0));
 		}
```

Comment:
Add more information about the request like device ID, request direction, etc? I think it would be better to carve out a function which dumps relevant information about a request.

Reply:
Done!